stable/redis/README.md

```diff
@@ -45,6 +45,35 @@ $ helm delete my-release
 
 The command removes all the Kubernetes components associated with the chart and deletes the release.
 
+## Upgrading an existing Release to a new major version
+
+A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an
+incompatible breaking change needing manual actions.
+
+### 4.0.0
+
+This version removes the `chart` label from the `spec.selector.matchLabels`
+which is immutable since `StatefulSet apps/v1beta2`. It has been inadvertently
+added, causing any subsequent upgrade to fail.
+
+It also update labels to match the recommendations from Helm:
+https://github.com/helm/helm/blob/master/docs/chart_best_practices/labels.md
+
+See https://github.com/helm/charts/pull/7686.
+
+In order to upgrade, delete the Redis master StatefulSet and slave Deployment before upgrading:
+```bash
+$ kubectl delete statefulsets.apps my-release-redis-master
```

Comment:
--cascade=false does not work here: after upgrade, the labels are so different that we have two pods for each kind, and we manually need to delete them. It may potentially cause data corruption if using the same PVC.

Reply:
@desaintmartin Are you saying that updating to the newer Helm-recommended labels is what's causing this issue? If so, perhaps this is something we can make a separate plan to address (in #7550, instead of all at once here)?

Reply:
Separating is easier (for us chart developers at least).